### PR TITLE
Luhn algorithm

### PR DESCRIPTION
**Pull request by QuantumNovice**
Perform Luhn validation on input string
    Algorithm:
    * Double every other digit starting from 2nd last digit.
    * Subtract 9 if number is greater than 9.
    * Sum the numbers
https://en.wikipedia.org/wiki/Luhn_algorithm

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by QuantumNovice** [comment]
mypy typing doesn't support mutable static types. I won't be changing that.
could we consider using `--allow-redefinition` flag?

File: DIRECTORY.md [hunk 1]
```diff
@@ -146,6 +146,7 @@
     * Number Theory
       * [Prime Numbers](https://github.com/TheAlgorithms/Python/blob/master/data_structures/hashing/number_theory/prime_numbers.py)
     * [Quadratic Probing](https://github.com/TheAlgorithms/Python/blob/master/data_structures/hashing/quadratic_probing.py)
+    * [Luhn's Algorithm](https://github.com/TheAlgorithms/Python/blob/master/data_structures/hashing/luhn.py)
```
Comment (onlinejudge95) [inline DIRECTORY.md:149]: Don't update this file, this is generated in the CI pipeline

> Do not update the README.md or DIRECTORY.md file which will be periodically autogenerated by our Travis CI processes.

**Review by onlinejudge95 — CHANGES_REQUESTED** [review]

File: hashes/luhn.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ...     79927398714, 79927398715, 79927398716, 79927398717, 79927398718,
+    ...     79927398719]
+    >>> test_cases = list(map(str, test_cases))
+    >>> list(map(is_luhn, test_cases))
+    [False, False, False, True, False, False, False, False, False, False]
+    """
+    check_digit: int
+    _vector: List[str] = list(string)
+    __vector, check_digit = _vector[:-1], int(_vector[-1])
+    vector: List[int] = [*map(int, __vector)]
```
[the remark this answers is not in the document]
Comment (QuantumNovice) [reply hashes/luhn.py:23]: mypy static checks doesn't pass when "the noise isn't there" :)

File: DIRECTORY.md (hunk 1 above)
Comment (QuantumNovice) [reply DIRECTORY.md:149]: my bad

**Review by onlinejudge95 — APPROVED** [review]

**Comment by onlinejudge95** [comment]
Nice contribution !!

**Pull request merged**